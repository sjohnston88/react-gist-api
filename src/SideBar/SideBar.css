.SideBar {
    width:20%;
    position: fixed;
}

.Blob {
  padding:10px 10px 5px 10px;
  text-align:left;
  cursor: pointer;
}

.italic {
    font-style: italic;
}

hr {
  margin: 10px 0 0 0;
}