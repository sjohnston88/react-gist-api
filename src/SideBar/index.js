import React from 'react';
import './SideBar.css';

class SideBar extends React.Component {

  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(id, e){
    e.preventDefault();
    this.props.preview(id)
  }

  render(){
    return (
      <div className="SideBar">
        {this.props.content.slice(0, 10).map(item => {
          return (
            <div key={item.id} className="Blob" onClick={(e) => this.handleClick(item.id, e)}>
              <b>{item.owner.login}</b>
              <br />
              <span className="italic">{item.description ? item.description : 'No Description'}</span>
              <hr />
            </div>
          ) 
        })}
      </div>
    );
  }
}

export default SideBar;