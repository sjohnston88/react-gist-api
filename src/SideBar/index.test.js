import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SideBar from './index';

Enzyme.configure({ adapter: new Adapter() });

describe('Component/SideBar', () => {
  const setup = () => shallow(<SideBar />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot('default');
  });
  
});
