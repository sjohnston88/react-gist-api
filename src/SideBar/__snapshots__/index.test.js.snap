// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <AppBar
          showMenuIconButton={false}
          title=""
          zDepth={1}
/>,
        <MenuItem
          anchorOrigin={
                    Object {
                              "horizontal": "right",
                              "vertical": "top",
                            }
          }
          checked={false}
          desktop={false}
          disabled={true}
          focusState="none"
          insetChildren={false}
          primaryText="Nothing to see here..."
          targetOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
/>,
      ],
      "disableSwipeToOpen": false,
      "docked": false,
      "onRequestChange": undefined,
      "open": null,
      "openSecondary": false,
      "style": Object {
        "paddingTop": "60px",
      },
      "swipeAreaWidth": 30,
      "width": null,
      "zDepth": 2,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "showMenuIconButton": false,
          "title": "",
          "zDepth": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorOrigin": Object {
            "horizontal": "right",
            "vertical": "top",
          },
          "checked": false,
          "desktop": false,
          "disabled": true,
          "focusState": "none",
          "insetChildren": false,
          "primaryText": "Nothing to see here...",
          "targetOrigin": Object {
            "horizontal": "left",
            "vertical": "top",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <AppBar
            showMenuIconButton={false}
            title=""
            zDepth={1}
/>,
          <MenuItem
            anchorOrigin={
                        Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
            }
            checked={false}
            desktop={false}
            disabled={true}
            focusState="none"
            insetChildren={false}
            primaryText="Nothing to see here..."
            targetOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
/>,
        ],
        "disableSwipeToOpen": false,
        "docked": false,
        "onRequestChange": undefined,
        "open": null,
        "openSecondary": false,
        "style": Object {
          "paddingTop": "60px",
        },
        "swipeAreaWidth": 30,
        "width": null,
        "zDepth": 2,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "showMenuIconButton": false,
            "title": "",
            "zDepth": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "right",
              "vertical": "top",
            },
            "checked": false,
            "desktop": false,
            "disabled": true,
            "focusState": "none",
            "insetChildren": false,
            "primaryText": "Nothing to see here...",
            "targetOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
